### ISSUE TITLE
feat: add custom log channels

### ISSUE THREAD

**Issue by phorcys420** [coder/coder#15239]
While talking with an user today, I realized that we don't have any way to dispatch logs from different processes spawned in the `startup_script` of an agent.

I think we should add a `log_channel` argument or something similar to `coder_script`, and each channel would have a tab next to build logs.
e.g right after cloning your project's repo, you would start `pnpm watch` or some other script to watch the backend.
```
data "coder_workspace" "me" {}

resource "coder_agent" "dev" {
  os   = "linux"
  arch = "amd64"
}

resource "coder_script" "backend" {
  agent_id           = coder_agent.dev.agent_id
  display_name       = "Backend"

  run_on_start       = true
  start_blocks_login = true

  script =  "cd projects/backend && pnpm watch"
  log_channel = true # or maybe change to a string, but I'm not sure this is necessary since we have display_name already
}

resource "coder_script" "frontend" {
  agent_id           = coder_agent.dev.agent_id
  display_name       = "Front-end"

  run_on_start       = true
  start_blocks_login = true

  script =  "cd projects/frontend && pnpm watch"
  log_channel = true # or maybe change to a string, but I'm not sure this is necessary since we have display_name already
}
```

I'm up to any suggestions since this is just what I came up with.
I think for proper UX we would also add a `coder logs attach` (or whatever) command in the future.

cc: @matifali @bpmct

**Comment by matifali** [comment]
I don't think I understand what is the purpose of this. We also have https://registry.terraform.io/providers/coder/coder/latest/docs/resources/script#log_path

**Comment by phorcys420** [comment]
@matifali so basically just a way to display app logs on the web ui instead of having to go to a file

**Comment by MrPeacockNLB** [comment]
yeah +1 from my side. we have more than 20 different scripts running at the same time which makes it difficult the find the logs you are interrested in. I do not know if splitting into tabs may solve this either. From UX point of view it may be cluttering then the UI. But we need a better way for displaying logs.

**Comment by matifali** [comment]
@chrifro what do you think about this?

**Comment by chrifro** [comment]
Would the custom logs live on the individual workspace page? 

I'm currently looking into polishing the individual workspace view (see [early draft](https://www.figma.com/design/OR75XeUI0Z3ksqt1mHsNQw/Dashbord-v1?node-id=263-1699&t=dDN5fLd6w7SUHwV6-1))  and with that I'll also look at how to improve the log view. So that would be perfect timing. I might come back to you @phorcys420 to get more details on this once I'm deeper into the logs view. 

Let me know in case you have anything else in mind.

**Comment by phorcys420** [comment]
> Would the custom logs live on the individual workspace page? 

yes, that's the idea!
basically the workspace logs would become a tabbed thing where you can add tabs for the logs you want to display in the web UI.

> I'll also look at how to improve the log view. So that would be perfect timing. I might come back to you @phorcys420 to get more details on this once I'm deeper into the logs view. 

Please do! -- just sent you a Slack
